point_pattern.py

```diff
@@ -119,7 +145,18 @@ def average_nearest_neighbor_distance(points):
      Measure of Spatial Relationships in Populations. Ecology. 35(4)
      p. 445-453.
     """
+    list_distance = []
     mean_d = 0
+    tot_mean = sum(points) / len(points)
+
+    for a in points:
+        d1 = euclidean_distance(x, y)
+        for b in points:
+            if a != b:
+                d2 = euclidean_distance(x, y)
+            if d1 <= d2:
+                list_distance.append(d1)
+                mean_d = sum(list_distance) / len(points)
 
```

Comment:
Overall this is on the right track.  You have the double for loop (to loop over the points and then loop over all other points) - good job.  You also have the check for self-neighbor, `if a != b` - also the right way to do this.

Each time you loop, in the outer loop, what if you stored the some really large distance, like `near = math.inf`.  Then, in the inner loop, after checking that `a != b`, you could compute the distance.  If the distance is less than `near`, you have to update `near`.  Do this for each point and keep a list of all of the nearest distances (`nearest = []` above all the loops and `nearest.append(near)` when you find the nearest neighbor for a given point).

If you like, update this method and commit again.